### ISSUE TITLE
SSL_CTX_load_verify_locations can succeed but leave a nonempty error stack

### ISSUE THREAD

**Issue by sfackler** [openssl/openssl#6851]
Using OpenSSL 1.1.0h, I'm seeing this behavior:

```
$ cat foo.c
#include <openssl/ssl.h>
#include <openssl/err.h>
#include <stdio.h>
#include <assert.h>

int main() {
	SSL_CTX *ctx = SSL_CTX_new(TLS_method());
	assert(ctx != NULL);
	printf("first load\n");
	assert(SSL_CTX_load_verify_locations(ctx, "/usr/local/etc/openssl@1.1/cert.pem", NULL) == 1);
	ERR_print_errors_fp(stdout);
	printf("second load\n");
	assert(SSL_CTX_load_verify_locations(ctx, "/usr/local/etc/openssl@1.1/cert.pem", NULL) == 1);
	ERR_print_errors_fp(stdout);
}
$ gcc foo.c -L /usr/local/opt/openssl@1.1/lib -lssl -lcrypto -lpthread -I /usr/local/opt/openssl@1.1/include
$ ./a.out
first load
second load
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
140735613809536:error:0B07C065:x509 certificate routines:X509_STORE_add_cert:cert already in hash table:crypto/x509/x509_lu.c:311:
```

~~It looks like #2830 cleared the error stack in by_dir.c, but not in by_file.c.~~ That change isn't in 1.1.0h

I noticed this because a later operation (I haven't tracked the exact function down) on the same thread after getting into this state failed with these errors. I'm not sure if the rest of the codebase can assume that a nonempty error stack means an error occurred, or if that's a separate bug.

**Comment by sfackler** [comment]
Ah, looks like this is roughly a duplicate of #2474, and #2830 is the fix for it. Are there plans to backport to 1.1.0?

**Comment by richsalz** [comment]
@paulidale do you want to backport your fix?

**Comment by paulidale** [comment]
Just for this problem or more widely from that PR?

**Comment by paulidale** [comment]
Fix is merged to 1.1.0.

**Comment by sfackler** [comment]
Thanks!
